### PR TITLE
Update more appliances and furniture to use pipe fittings, where they make sense.

### LINKED ISSUE #44323: Pipe fittings are too rare in the world, should be recoverable from more things
# Is your feature request related to a problem? Please describe.
At the moment, Pipe fittings are only acquirable from the following sources: workbench, muffler, seat, leather seat, electric kiln, plow

4 of those elements require access to vehicles and the ability to disassemble seats, which is a strange bar. There are lots of other things in the environment that should give these when disassembled, such as the furnace, hot water tank, washer, fridge and sink.

# Describe the solution you'd like

Several items should already need pipe fittings during destruction/deconstruction. A small, non-exhaustive list follows:

- [x] Fridge
- [x] Glass Door Fridge
- [x] Freezer
- [x] Sink
- [x] Furnace
- [x] Hot Water Heater
- [x] Pump
- [x] High-gauge pump
- [x] Cooling Unit
- [x] Heating unit
- [x] Satelite Dish
- [x] Cell Phone Signal Booster
- [ ] Still
- [x] Monkey Bars
- [x] Park Slide
- [ ] All engines(even through the use of rubber hoses, there still should be usable coupling components)
- [ ] Fuel tanks
- [x] Stove
- [x] Rotary Clothes Dryer Line
- [x] Shower
- [ ] Hydraulic Press
- [x] Chain Link Fence(The top bar would still be connecting to the metal posts through T fittings)
- [x] Dishwasher
- [x] Floor Lamp
- [ ] Skate Park Railing

# Describe alternatives you've considered

- Removing pipe fittings in exchange for a required welding material of some kind( eg PVC glue)

### PR DESCRIPTION
**Pull request by Nymall**
#### Summary

SUMMARY: Balance "Update more appliances and furniture to use pipe fittings, where they make sense."

#### Purpose of change

Updating more furniature to give back pipe fittings when broken, where it seems to make sense.

Fixes #44323 

Please see issue #44323 

#### Describe the solution

Added pipe fittings in following amounts to following entities:

f_home_furnace (2-6)
f_oven (1-3)
f_server (2-4)
f_sattelite (4-8)
f_solar_unit (1-3)
f_rotart_clothesline (1-3)
f_floor_lamp (1), increased scrap by 1
f_machinery_heavy (1-3)
f_machinery_old (1-2)
f_sample_freezer (2-6), bring in line with f_fridge
f_lab_bench (4-8), bring in line with f_workbench
f_eyewash (0-1)
f_IV_Pole (1)
f_bathtub (0-2, bash only)
f_shower (0-2, bash only)
f_sink (0-2, bash only)
f_water_heater (2-6)
f_water_purifier (2-4)
f_ergometer (2-6, bash 0-4)
f_ergometer_mechanical (2-6, bash 0-4)
f_treadmill (2-4)
f_treadmill_mechanical (2-4)
f_cellphone_booster (1-2), also added Pipe(1)
f_small_satelite_dish (1-2), also added pipe(1-4)
f_tv_antenna (1-4)
f_air_compressor (2-4)
f_aut_gas_console_o (1), and by extension f_aut_gas_console
t_gas_pump_smashed (2-6), also added pipe(4-8)
t_diesel_pump_smashed (2-6), also added pipe(4-8)
t_sewage_pump (0-4)
t_water_pump (0-1)
t_water_dispenser (0-1)
t_milking_machine (1-2)
t_bulk_tank (1-2)
t_slide (2-4)
t_monkey_bars (4-8)

#### Describe alternatives you've considered

Proposing removal of pipe fitting instead of updating, or changing to something like PVC Glue.

### CODE REVIEW

**Comment by Nymall** [comment]
This is super frustrating - I have no idea what I'm doing wrong now. The validation output is showing three issues and seven files that need to be linted, however whenever I try to make any updates or changes the files un-lint. I can't get visual studio up and running - is there anything else I can use for bulk linting that does not need to be built?

**Comment by Nymall** [comment]
I'm at a complete loss. The fix should be super simple, but it's not jumping out at me. It dosen't help that the error messages from "JSON Validation / JSON style check" are fairly unhelpful. I'm not quite sure what is triggering this issue - any help would be deeply appreciated.

File: data/json/furniture_and_terrain/furniture-roof.json [hunk 1]
```diff
@@ -82,9 +86,7 @@
     "move_cost_mod": 2,
     "required_str": 8,
     "flags": [ "TRANSPARENT", "BASHABLE" ],
-    "deconstruct": {
-      "items": [ { "item": "pipe", "count": 6 }, { "item": "steel_chunk", "count": 2 }, { "item": "cable", "charges": 10 } ]
-    },
+    "deconstruct": { "items": [ { "item": "pipe", "count": 6 }, { "item": "steel_chunk", "count": 2 }, { "item": "pipe_fittings", "count": [ 1, 4 ] }, { "item": "cable", "charges": 10 } ] },
```
Comment (UmbralReaper) [inline data/json/furniture_and_terrain/furniture-roof.json:89]: "deconstruct": {
      "items": [
        { "item": "pipe", "count": 6 },
        { "item": "steel_chunk", "count": 2 },
        { "item": "pipe_fittings", "count": [ 1, 4 ] },
        { "item": "cable", "charges": 10 }
      ]
    },

Comment (UmbralReaper) [reply data/json/furniture_and_terrain/furniture-roof.json:89]: Check with the web linter, http://dev.narc.ro/cataclysm/format.html

Comment (Nymall) [reply data/json/furniture_and_terrain/furniture-roof.json]: Ran through it again - I've already checked with the web linter(second commit, 4th and 5th). Not quite sure what's going wrong. Going to repost file directly into github web from the linter.

Comment (Nymall) [reply data/json/furniture_and_terrain/furniture-roof.json]: This time it worked. Guess I'm going to have to use the web interface from here on out. Thank you!

**Review by I-am-Erk — CHANGES_REQUESTED** [review]
I don't think we should be using pipe fittings to represent "every place pipes are attached to something". I'd suggest keeping them limited to situations where the pipe fitting is part of a fluid system. Pipe attachment points don't require a special item, you can make them from just about anything.

**Comment by Nymall** [comment]
Certainly. So would you like the following to be removed?

f_machinery_heavy (1-3)
f_machinery_old (1-2)
f_lab_bench (4-8), bring in line with f_workbench
f_IV_Pole (1)
f_ergometer (2-6, bash 0-4)
f_ergometer_mechanical (2-6, bash 0-4)
f_treadmill (2-4)
f_treadmill_mechanical (2-4)
f_cellphone_booster (1-2), also added Pipe(1)
f_small_satelite_dish (1-2), also added pipe(1-4)
f_tv_antenna (1-4)
t_slide (2-4)
t_monkey_bars (4-8)

**Comment by I-am-Erk** [comment]
f_machinery_heavy (1-3)
f_machinery_old (1-2)
f_lab_bench (4-8), bring in line with f_workbench

Those three could all keep them. The rest are good ones to remove yeah

**Comment by Nymall** [comment]
Updated and removed offending items. Thank you!

**Comment by kevingranade** [comment]
Neds json linting

**Pull request merged**